docs/advanced.rst

```diff
@@ -483,8 +476,92 @@ The full syntax for the :attr:`sanity_patterns` is the following:
       sn.assert_eq(sn.count(numbers), 100),
       sn.all(sn.map(lambda x: sn.assert_bounded(x, 90, 100), numbers)))
 
-Note that the environment variables ``LOWER`` and ``UPPER`` have to be exported before execution of the ``advanced_example6.sh`` script.
-Within ReFrame it is possible to define commands that will be run before execution of the actual :attr:`executable <reframe.core.pipeline.RegressionTest.executable>`.
-To achieve this, you can use the :attr:`pre_run <reframe.core.pipeline.RegressionTest.pre_run>` attribute of the :class:`RegressionTest`.
-The :attr:`pre_run <reframe.core.pipeline.RegressionTest.pre_run>` attribute is a list of shell commands to be emitted verbatim in the generated job script before the executable.
-In this case, we make sure that the ``limits.sh`` file is sourced before executing the ``advanced_example6.sh`` script.
+
+Customizing the Generated Job Script
+------------------------------------
+
+It is often the case that you must run some commands before and/or after the parallel launch of your executable.
+This can be easily achieved by using the :attr:`pre_run <reframe.core.pipeline.RegressionTest.pre_run>` and :attr:`post_run <reframe.core.pipeline.RegressionTest.post_run>` attributes of :class:`RegressionTest`.
+
+The following example is a slightly modified version of the previous one.
+The lower and upper limits for the random numbers are now set inside a helper shell script in ``scripts/limits.sh`` and we want also to print the word ``FINISHED`` after the parallel job has finished.
```

Comment:
Maybe refer to the job as 'submitted' instead of 'parallel'.

Reply:
Here, I mean the executable, so "submitted job" is more confusing, I think. I'll see how I can rephrase it.